### PR TITLE
Progress Bar

### PR DESCRIPTION
**Pull request by brenpardal**
### **WHAT'S NEW**
**This update includes:**
* Added a progress bar to main_activity layout file

#### *Preview ProgressBar added:*
progress_bar

### CODE REVIEW

File: app/src/main/java/com/weatherapp/mvp/view/WeatherView.kt [hunk 1]
```diff
@@ -18,14 +19,24 @@ class WeatherView(activity: Activity, private val binding: ActivityMainBinding)
     override fun showData(weatherFiltered: List<WeatherData>, item: ItemClicked) {
         binding.recyclerViewWeather.layoutManager = LinearLayoutManager(context)
         binding.recyclerViewWeather.adapter = WeatherAdapter(weatherFiltered, item)
+        hideProgressBar()
```
Comment (mauroLonghi) [inline app/src/main/java/com/weatherapp/mvp/view/WeatherView.kt:22]: pls change this call on the presenter, after you call to show data.

File: app/src/main/java/com/weatherapp/activity/MainActivity.kt [hunk 2]
```diff
@@ -19,6 +19,7 @@ class MainActivity : AppCompatActivity(), ItemClicked {
         binding = ActivityMainBinding.inflate(layoutInflater)
         setContentView(binding.root)
         presenter = WeatherPresenter(WeatherModel(WeatherService()), WeatherView(this, binding))
+        presenter.showLoading()
```
Comment (SartiMau) [inline app/src/main/java/com/weatherapp/activity/MainActivity.kt:22]: this should not be here.. you have to call the showLoading from the presenter

File: app/src/main/java/com/weatherapp/mvp/presenter/WeatherPresenter.kt [hunk 3]
```diff
@@ -21,4 +21,12 @@ class WeatherPresenter(
     override fun itemWeatherClicked(date: String) {
         view.openDetailsFragment(date, model.weatherList)
     }
+
+    override fun showLoading() {
+        view.showProgressBar()
+    }
+
+    override fun hideLoading() {
```
Comment (SartiMau) [inline app/src/main/java/com/weatherapp/mvp/presenter/WeatherPresenter.kt:29]: I think that these two methods are not necessary as you will only call view.showProgressBar() and view.hideProgressBar() from the presenter

File: app/src/main/java/com/weatherapp/mvp/view/WeatherView.kt [hunk 4]
```diff
@@ -18,14 +19,24 @@ class WeatherView(activity: Activity, private val binding: ActivityMainBinding)
...
     }
 
     override fun showError() {
+        hideProgressBar()
```
Comment (SartiMau) [inline app/src/main/java/com/weatherapp/mvp/view/WeatherView.kt:26]: same here

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/weatherapp/mvp/view/WeatherView.kt (hunk 1 above)
Comment (brenpardal) [reply app/src/main/java/com/weatherapp/mvp/view/WeatherView.kt]: Solved!

File: app/src/main/java/com/weatherapp/activity/MainActivity.kt (hunk 2 above)
Comment (brenpardal) [reply app/src/main/java/com/weatherapp/activity/MainActivity.kt]: Solved!

File: app/src/main/java/com/weatherapp/mvp/presenter/WeatherPresenter.kt (hunk 3 above)
Comment (brenpardal) [reply app/src/main/java/com/weatherapp/mvp/presenter/WeatherPresenter.kt]: Yes, solved!

File: app/src/main/java/com/weatherapp/mvp/view/WeatherView.kt (hunk 4 above)
Comment (brenpardal) [reply app/src/main/java/com/weatherapp/mvp/view/WeatherView.kt]: Solved!

**Review by SartiMau — APPROVED** [review]

**Review by nicolbidone — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Pull request merged**